src/components/interactive-builder/edit-question.modal.tsx

```diff
@@ -809,15 +864,131 @@ const EditQuestionModal: React.FC<EditQuestionModalProps> = ({
                         titleText={t('selectAnswersToDisplay', 'Select answers to display')}
                       />
                     ) : null}
-                    {(hasConceptChanged ?? answersChanged) && (
+
+                    {hasConceptChanged && selectedAnswers.length && questionToEdit.type !== 'programState' ? (
                       <div>
-                        {selectedAnswers.map((selectedAnswer) => (
-                          <Tag className={styles.tag} key={selectedAnswer.id} type={'blue'}>
-                            {selectedAnswer.text}
+                        {selectedAnswers?.map((answer) => (
+                          <Tag className={styles.tag} key={answer?.id} type={'blue'}>
+                            {answer?.text}
                           </Tag>
                         ))}
                       </div>
-                    )}
+                    ) : null}
+
+                    {(selectedConcept || questionToEdit) && questionToEdit?.questionOptions.answers?.length ? (
```

Comment:
Because of the condition `questionToEdit?.questionOptions.answers?.length`, if a user when creating a question doesn't select any answer, they wouldn't be able to add any answers when editing either, and that shouldn't be the case.

Reply:
@NethmiRodrigo  I found a way to do it kindly check and advise

Reply:
In the add-question-modal, this condition makes sense. In the edit version, if the user has not selected any answer during creation, they should still be able to add answers. What I meant was that we should just remove this condition and show the `Add more answers` if the concept is of the datatype `coded`

Reply:
@NethmiRodrigo how do we check if the datatype is coded? Because even in add question, the check is done to see if the concept has answers there is no check for the datatype

Reply:
So in the `add-question-modal`, the reason that there is no check for the datatype is because as soon as you search for a concept from the API and select a concept, if the concept has answers, it got added to a state, so the check for the datatype `coded` wasn't necessary (hacky way to do it) - https://github.com/openmrs/openmrs-esm-form-builder/blob/5154c460ff533c4bec4180486e92cc5bda168b05/src/components/interactive-builder/add-question.modal.tsx#L160-L170